File: solutions/erlang/armstrong-numbers/1/src/armstrong_numbers.erl
-module(armstrong_numbers).

-export([is_armstrong_number/1]).


is_armstrong_number(Number) -> 
  Digits = lists:map(fun (X) -> list_to_integer([X]) end, integer_to_list(Number)),
  Exponent = length(Digits),
  Sum = lists:sum(lists:map(fun (X) -> math:pow(X, Exponent) end, Digits)),
  Sum == Number.

File: solutions/lua/armstrong-numbers/1/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
  if number == 0 then
    return 0 == 0 ^ 1
  end

  local s = tostring(number)
  local count = #s

  local digits = {}
  local sum = 0
  for i = 1, count do
    local d = s:byte(i) - string.byte('0')
    sum = sum + (d ^ count)
  end

  return sum == number
end

return ArmstrongNumbers
